### ISSUE TITLE
length does not error when evaluating a non-existing element

### ISSUE THREAD

**Issue by bnoorduin** [mikefarah/yq#683]
**Describe the bug**
When piping eval of non-existing element through length, do not get error - result is '4'.

Version of yq: 4.4.1
Operating system: linux
Installed via: 
curl -sSL -o yq https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 && chmod +x yq

**Input Yaml**
vault-config.yml:
```yaml
vault:
  policies:
    - name: admin
      file: policy-admin.hcl
    - name: gitlab
      file: policy-gitlab.hcl
    - name: gitlab-operations-standard
      file: policy-gitlab-operations-standard.hcl
    - name: gitlab-operations-protected
      file: policy-gitlab-operations-protected.hcl
```

**Command**
The command you ran:
```
yq eval -e '.vault.test | length' vault-config.yml
```

**Actual behavior**

```yaml
4
```

**Expected behavior**

```yaml
0
Error: no matches found
```

**Additional context**
I would expect either '0' return or error return. Certainly not '4'.

For work-around I check whether the element exists before. But I would expect with -e the evaluation to fail.

```bash
$ >/dev/null 2>&1 yq eval -e '.vault.test' vault-config.yml || die "no such element '.vault.test' defined"
```

**Comment by mikefarah** [comment]
Looks like it's returning the length of the string `null` 🤦  - will fix in the next release

**Comment by mikefarah** [comment]
Note, this will be fixed to return 0 for null values / non existant values

**Comment by mikefarah** [comment]
Fixed in 4.5.0!

**Comment by bnoorduin** [comment]
Awesome!!
